### PR TITLE
[WIP]Add ProPainter

### LINKED ISSUE #26360: Add ProPainter to transformers
### Model description

ProPainter can be an excellent addition to Transformers given its impressive inpainting capabilities. 

*Relevant Links*
Paper - https://arxiv.org/abs/2309.03897
Project Page - https://shangchenzhou.com/projects/ProPainter/
Original Code - https://github.com/sczhou/ProPainter
Weights - https://github.com/sczhou/ProPainter/releases/tag/v0.1.0 (Needs to be merged into a unified one)

Author - @sczhou 

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

@amyeroberts @ArthurZucker
If you think this is a valuable addition I'm more than happy to work on it.

### PR DESCRIPTION
**Pull request by shauray8**
# What does this PR do?

Adds ProPainter to transformers.
Author - https://github.com/sczhou/ProPainter
hub - https://huggingface.co/shauray/ProPainter-hf/

Fixes https://github.com/huggingface/transformers/issues/26360
*issue contains all the other relevant links*

## Before submitting
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@rafaelpadilla @LysandreJik

### CODE REVIEW

File: src/transformers/models/propainter/__init__.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Copyright 2021 The HuggingFace Team. All rights reserved.
```
Comment (rafaelpadilla) [inline src/transformers/models/propainter/__init__.py:1]:
```suggestion
# Copyright 2023 The HuggingFace Team. All rights reserved.
```

**Comment by rafaelpadilla** [comment]
Hi @shauray8 :)

Please, let me know when it is ready for review

**Comment by shauray8** [comment]
I'll ping you before the EOD.

**Comment by mahimairaja** [comment]
Hi, I have added few codes for Image Processing here https://github.com/huggingface/transformers/pull/26449

I checkedout out this PR #26391 before modifying but when I opened a PR, happened to open a new 

 Is there anyway I can merge #26449 with this one?

**Comment by amyeroberts** [comment]
@shauray8 Any update on adding this model?

**Comment by shauray8** [comment]
@amyeroberts it's almost done, I'm stuck with tests rn and I might be able to push it for 1st review tomorrow.

**Comment by amyeroberts** [comment]
@shauray8 Great! Let us know if you need any help with the tests.

**Comment by royvelich** [comment]
What happens with this PR? :-)

**Comment by amyeroberts** [comment]
@royvelich This PR became inactive and was then closed automatically. If you or anyone else in the community would like to open another PR to add this model we'd be happy to review!

**Comment by RUFFY-369** [comment]
@amyeroberts Can I take this on?! As I'm already looking into it. :)

**Comment by amyeroberts** [comment]
@RUFFY-369 Sure!

**Pull request closed without merging**